lib/modules/dosomething/dosomething_api/includes/Transformer.php

```diff
@@ -217,16 +217,123 @@ protected function transformCollection($items, $method = 'transform') {
    *   An object containing properties of Campaign data:
    *   - nid: (string) Campaign id.
    *   - title: (string) Campaign title.
+   *   - tagline: (string) Campaign tagline.
+   *   - created: (string) Timestamp when campaign created.
+   *   - update: (string) Timestamp of most recent update to campaign.
+   *   - active_hours: (string) Average number of hours campaign takes to complete.
+   *   - cover_image:
+   *   - cover_image_alt:
+   *   - scholarship:
+   *   - is_staff_pick:
+   *   - fact_problem:
+   *   - fact_solution:
+   *   - fact_sources:
+   *   - solution:
+   *   - primary_cause:
+   *   - secondary_cause:
    *
    * @return array
    */
   protected function transformCampaign($data) {
-    return array(
-      'campaign' => array(
-        'id' => $data->nid,
-        'title' => $data->title,
-      ),
+    $output = array(
+      'id' => $data->id,
+      'title' => $data->title,
     );
+
+    if ($data instanceof Campaign) {
+      $output['tagline'] = $data->tagline;
+      $output['created_at'] = $data->created_at;
+      $output['updated_at'] = $data->updated_at;
+      $output['time_commitment'] = $data->time_commitment;
+      // $output['type'] = $data->type; //@TODO: Should type be included? Consider there is an SMS Campaign type...
+
+      if ($data->status) {
+        // @TODO: Should the status default to "active"?
+        $output['status'] = $data->status;
+      }
+
+      if ($data->cover_image) {
+        foreach ($data->cover_image as $key => $image) {
+          if (!is_null($image)) {
+            $output['cover_image'][$key] = $this->transformMedia($image, 'square');
+          }
+        }
+      }
+
+      if ($data->facts['problem']) {
+        $output['facts']['problem'] = $data->facts['problem']['fact'];
+      }
+
+      if ($data->facts['solution']) {
+        $output['facts']['solution'] = $data->facts['solution']['fact'];
+      }
+
+      if ($data->facts['sources']) {
+        $output['facts']['sources'] = $data->facts['sources'];
```

Comment:
ohh. hm. yeah that sounds like it'd break stuff.  Umm ideally we should update the fact stuff to be consistent, because from the API perspective, i'd expect this to not be HTML since it doesn't have that array. I don't think it should block merging this PR, perhaps just a new issue?